src/x25519.rs

```diff
@@ -76,14 +76,33 @@ impl EphemeralSecret {
         SharedSecret(self.0 * their_public.0)
     }
 
-    /// Generate an x25519 [`EphemeralSecret`] key.
+    /// Generate an x25519 [`EphemeralSecret`] key with the supplied [`rand_core::OsRng`]
+    #[deprecated(
+        since = "2.0.0",
+        note = "Renamed to `random_from_rng`. This will be removed in 2.1.0"
+    )]
     pub fn new<T: RngCore + CryptoRng>(mut csprng: T) -> Self {
         let mut bytes = [0u8; 32];
 
         csprng.fill_bytes(&mut bytes);
 
         EphemeralSecret(Scalar::from_bits_clamped(bytes))
     }
+
+    /// Generate an x25519 [`EphemeralSecret`] key with the supplied [`rand_core::OsRng`]
```

Comment:
Ditto. This is repeated 4 or 5 more times, so I'll just omit the dittos

Reply:
Yeah I just took it out and simplified it since the rendered documentation links them anyways.